asmcli/asmcli.sh

```diff
@@ -1935,22 +1960,34 @@ generate_membership_name() {
 register_cluster() {
   if is_cluster_registered; then return; fi
 
-  if can_modify_gcp_components; then
-    enable_workload_identity
-  else
-    exit_if_no_workload_identity
+  if is_gcp; then
+    if can_modify_gcp_components; then
+      enable_workload_identity
+    else
+      exit_if_no_workload_identity
+    fi
+    populate_cluster_values
   fi
 
-  populate_cluster_values
-  local MEMBERSHIP_NAME
-  MEMBERSHIP_NAME="$(generate_membership_name)"
+  local MEMBERSHIP_NAME; MEMBERSHIP_NAME="$(generate_membership_name)"
   info "Registering the cluster as ${MEMBERSHIP_NAME}..."
-
   local FLEET_ID; FLEET_ID="$(context_get-option "FLEET_ID")"
-  retry 2 gcloud beta container hub memberships register "${MEMBERSHIP_NAME}" \
-    --project="${FLEET_ID}" \
-    --gke-uri="${GKE_CLUSTER_URI}" \
-    --enable-workload-identity
+
+  if is_gcp; then
+    retry 2 gcloud beta container hub memberships register "${MEMBERSHIP_NAME}" \
+      --project="${FLEET_ID}" \
+      --gke-uri="${GKE_CLUSTER_URI}" \
+      --enable-workload-identity
+  else
+    local KCF; KCF="$(context_get-option "KUBECONFIG")"
+    local KCC; KCC="$(context_get-option "CONTEXT")"
+
+    retry 2 gcloud beta container hub memberships register "${MEMBERSHIP_NAME}" \
+      --project="${FLEET_ID}" \
+      --kubeconfig="${KCF}" \
+      --context="${KCC}" \
+      --enable-workload-identity
+  fi
```

Comment:
In other places we've used a command-builder pattern for this, maybe something like:

```
local CMD
CMD="gcloud beta container hub memberships register ${MEMBERSHIP_NAME}"
CMD="${CMD} --project=${FLEET_ID}"
CMD="${CMD} --enable-workload-identity"
if is_gcp; then
  CMD="${CMD} --gke-uri=${GKE_CLUSTER_URI}"
else
  CMD="${CMD} --kubeconfig=${KCF} --context=${KCC}"
fi
retry 2 run "${CMD}"
```

Reply:
thanks for the exact snippet!